samples/msgext-search-sso-config/nodejs/bots/teamsMessagingExtensionsSearchAuthConfigBot.js

```diff
@@ -229,23 +232,25 @@ class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
     // Overloaded function. Receives invoke activities with the name 'composeExtension/fetchTask'
     async handleTeamsMessagingExtensionFetchTask(context, action) {
         if (action.commandId === 'SHOWPROFILE') {
+            const userTokenClient = context.turnState.get(context.adapter.UserTokenClientKey);
             const magicCode =
-                action.state && Number.isInteger(Number(action.state))
-                    ? action.state
+                context.state && Number.isInteger(Number(context.state))
+                    ? context.state
                     : '';
-            const tokenResponse = await context.adapter.getUserToken(
-                context,
+
+            const tokenResponse = await userTokenClient.getUserToken(
+                context.activity.from.id,
                 this.connectionName,
+                context.activity.channelId,
                 magicCode
             );
 
             if (!tokenResponse || !tokenResponse.token) {
                 // There is no token, so the user has not signed in yet.
                 // Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions
-
-                const signInLink = await context.adapter.getSignInLink(
-                    context,
-                    this.connectionName
+                const { signInLink } = await userTokenClient.getSignInResource(
```

Comment:
Here I tried, but show "unsupported"
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/37bddac9-a2b3-46f1-8969-562ab21efa38)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/d41d9e78-cabc-4f81-a4e3-3a2f8afa5417)

Reply:
Here I tried  - 
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/e0b1b21d-201d-465b-8fa2-326cdc84e900)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/03244380-54b1-4502-b9d2-82bb0a30acfd)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/b4fbda23-ff72-494d-b9f8-979d8ee708b8)

Reply:
yes, you tried this work because you consent already. 
I met the `unsupported` failure is not consent yet.

Reply:
Yes correct @wenytang-ms,. Please give consent and check its working or not.

Global Administrator can grant consent using following link: https://login.microsoftonline.com/common/adminconsent?client_id=<%appId%>

Reply:
So if this compose action requires user consent before, then why do we need to return a signInLink action next?

Reply:
The first time you click login, that time signInLink returns the link and opens the URL.

![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/f8796bde-3392-4804-abf1-3461c104dfc3)

Reply:
For example, if you create an app registration for a different tenant, you're testing a different tenant. At that time, we need to give the grant administrator consent.

Global Administrator can grant consent using following link: https://login.microsoftonline.com/common/adminconsent?client_id=<%appId%>

![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/c68e2353-6de4-4af9-a7e1-a4ec5a383826)